docs/source/grpo_trainer.md

```diff
@@ -529,6 +529,144 @@ and the reward will be computed as the sum of the rewards from each function, or
+            yield {
+                "prompt": prompt,
+                "image": np.random.uniform(low=0.0, high=255.0, size=(224, 224, 3)).astype(np.uint8),
+            }
+    
+    return Dataset.from_generator(
+        data_gen, 
+        gen_kwargs={"num_samples": 16}, 
+        features=Features(image=Image(), prompt=Value(dtype="string"))
+    )
+
+# Configure model with quantization
+quantization_config = BitsAndBytesConfig(
+    load_in_4bit=True,
+    bnb_4bit_compute_dtype="bfloat16",
+    bnb_4bit_quant_type="nf4",
+    bnb_4bit_use_double_quant=True,
+)
+
+model = AutoModelForImageTextToText.from_pretrained(
+    "Qwen/Qwen2-VL-2B-Instruct",
+    torch_dtype="bfloat16",
+    quantization_config=quantization_config,
+)
+
+processor = AutoProcessor.from_pretrained("Qwen/Qwen2-VL-2B-Instruct")
+
+# Define reward function for VLM
+def vlm_reward_func(prompts, completions, **kwargs):
+    # Custom reward logic for VLM outputs
+    return [-((len(c) - 25) ** 2) + 100 for c in completions]
+
+# Configure LoRA for efficient training
+lora_config = LoraConfig(
+    task_type="CAUSAL_LM",
+    r=8,
+    lora_alpha=32,
+    lora_dropout=0.1,
+    target_modules=["q_proj", "v_proj"],
+)
+
+# Training configuration
+training_args = GRPOConfig(
+    output_dir="./vlm_grpo_output",
+    learning_rate=1e-5,
+    per_device_train_batch_size=2,
+    num_generations=4,
+    max_completion_length=32,
+    max_prompt_length=None,  # Allow full length for image tokens
+    use_vllm=True,  # Enable vLLM acceleration
+    vllm_mode="colocate",  # Use colocate mode
+    vllm_gpu_memory_utilization=0.3,
+    report_to="wandb",
+)
+
+# Create trainer
+trainer = GRPOTrainer(
+    model=model,
+    processing_class=processor,  # VLM processor
+    reward_funcs=[vlm_reward_func],
+    args=training_args,
+    train_dataset=create_vlm_dataset(),
+    peft_config=lora_config,
+)
+
+# Start training
+trainer.train()
+```
+
+### VLM-Specific Configuration
+
+When training VLMs, consider these configuration options:
+
+- **`max_prompt_length=None`**: Set to `None` to avoid truncating image tokens
+- **`vllm_gpu_memory_utilization`**: Reduce if using large VLMs (e.g., 0.2-0.3)
+- **LoRA Configuration**: Target vision-language projection layers if needed
+- **Quantization**: 4-bit quantization significantly reduces memory usage
+
+### Performance Considerations
+
+- **Memory Usage**: VLMs require more memory due to image processing. Use quantization and LoRA for large models
+- **vLLM Support**: VLM models work with vLLM v0.6.1+ in both server and colocate modes
+- **Batch Size**: Start with smaller batch sizes due to increased memory requirements from images
+- **Generation Length**: Consider longer max completion lengths for detailed image descriptions
+
+### Dataset Format
+
+VLM datasets should include:
+- **`prompt`**: Text prompt formatted through the processor's chat template
+- **`image`**: PIL Image or numpy array representing the image
```

Comment:
We could add some details about the image support since currently only 1 image is supported per prompt

```suggestion
- **`image`**: PIL Image or numpy array representing the image (only one image per prompt is supported)
```